project-1/albertafriadii/pkg/books/repository/book_repo.go

```diff
@@ -0,0 +1,194 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"golang-mentoring/project-1/albertafriadii/pkg/config"
+	"golang-mentoring/project-1/albertafriadii/pkg/domain"
+
+	"gorm.io/gorm"
+)
+
+type BookDBRepository struct {
+	DB *gorm.DB
+}
+
+func NewBookDBRepository(db *gorm.DB) domain.BookRepositoryInterface {
+	return &BookDBRepository{
+		DB: db,
+	}
+}
+
+func (r *BookDBRepository) ListBook(ctx context.Context) ([]domain.Book, error) {
+	var book []domain.Book
+
+	err := r.DB.WithContext(ctx).Find(&book).Order("created_at DESC").Error
+	if err != nil {
+		return nil, fmt.Errorf("BookDBRepository.ListBook: %w", err)
+	}
+
+	if book == nil || len(book) == 0 {
+		return nil, config.ErrNotFound
+	}
+
+	return book, nil
+}
+
+func (r *BookDBRepository) GetBook(ctx context.Context, BookID uint) (*domain.Book, error) {
+	var book domain.Book
+
+	err := r.DB.WithContext(ctx).First(&book, "book_id = ?", BookID).Error
+	if err != nil {
+		return nil, fmt.Errorf("BookDBRepository.GetBook: %w", err)
+	}
+
+	return &book, nil
+}
+
+func (r *BookDBRepository) GetBookByUserID(ctx context.Context, BookID uint, UserID uint) (*domain.Book, error) {
+	var book domain.Book
+
+	err := r.DB.Table("borrowbooks").Select("books.book_id, books.title, books.author, books.publisher, books.summary, books.stock, books.max_stock").Joins("INNER JOIN users ON borrowbooks.user_id = users.user_id").Joins("INNER JOIN books ON borrowbooks.book_id = books.book_id").Where("users.user_id = ? AND borrowbooks.book_id = ?", UserID, BookID).Find(&book).Error
```

Comment:
line terlalu panjang, dijadikan beberapa line
```suggestion
	err := r.DB.Table("borrowbooks").
	Select("books.book_id, books.title, books.author, books.publisher, books.summary, books.stock, books.max_stock").
	Joins("INNER JOIN users ON borrowbooks.user_id = users.user_id").
	Joins("INNER JOIN books ON borrowbooks.book_id = books.book_id").
	Where("users.user_id = ? AND borrowbooks.book_id = ?", UserID, BookID).
	Find(&book).Error
```

Reply:
Sama itu nama tabelnya nggak salah? di file SQL migration-nya, nama tabelnya ini
![Screenshot_20220919_175500](https://user-images.githubusercontent.com/12052131/191002522-de103fdb-bcdc-4aad-87e1-79f33b2ffcad.png)